File: 2020/07/12-AddDigits.cpp
class Solution {
public:
    int addDigits(int num) {
        int n = num;
        
        while (n > 9) {
            int sum = 0;
            while (n > 0) {
                sum = sum + (n % 10);
                n /= 10;
            }
            n = sum;
        }
        
        return n;
    }
};

File: 2020/07/12-AddDigits.kt
class Solution {
    fun addDigits(num: Int): Int {
        var n = num
        
        while (n > 9) {
            var sum = 0
            while (n > 0) {
                sum = sum + (n % 10)
                n /= 10
            }
            
            n = sum
        }
        
        return n
    }
}